### ISSUE TITLE
Interfaces

### ISSUE THREAD

**Issue by mikeric** [mikeric/rivets#178]
As a follow-up to #92, I think the keypath access notations (`.` and `:`) should not be hardcoded into Rivets.js core, it's confusing to beginners what these are for, and "why" you actually need to define an adapter. Instead they should be exposed as interfaces that you can add and overwrite however you like, but not required to start using Rivets.js.

By default, Rivets.js would ship with a single interface for the `.` notation to observe properties on POJSOs using ES5 natives. This interface will work on all ES5-compliant browsers (IE9+, FF4+, SF5+, CH7+ and OP12+).

If you're targeting non-ES5 browsers, you can overwrite the default interface to work with an evented model library of your choice that has the legacy browser support you need.

You can add as many interfaces as you like and assign whatever separator you want to them. For example, you can keep `.` for subscribing to plain objects, add `:` or `@` for subscribing to Backbone.js model attributes or even `#` for subscribing to validation errors if you want.

The API for defining an interface would be similar as for defining the adapter in `< 0.6.0`.

``` javascript
rivets.interfaces[":"] = {
  observe: function(keypath, callback) {
    this.on("change:" + keypath, callback)
  },
  unobserve: function(keypath, callback) {
    this.off("change:" + keypath, callback)
  }
  get: function(keypath) {
    return this.get(keypath)
  },
  set: function(keypath, value) {
    this.set(keypath, value)
  }
}
```

There would be no more `:` notation by default. If you want similar functionality as the adapter bypass, you can just implement an interface for it (leave out the `observe` and `unobserve` functions and define a simple `get` and `set`).

``` javascript
rivets.interfaces[":"] = {
  get: function(keypath) {
    return this[keypath]
  },
  set: function(keypath, value) {
    this[keypath] = value
  }
}
```

To sum things up, there would now be 3 main concepts for tuning Rivets.js to your application.
- **Interfaces:** Defines how Rivets.js observes and interacts with the model / keypath.
- **Binders:** Defines how Rivets.js affects the DOM and/or model when properties change.
- **Formatters:** Defines how Rivets.js affects incoming values before it goes through the binder and/or back to the model.

All of which are accessible in the universal syntax for binding declarations:

```
data-[prefix]-[binder]="[model][interface][keypath] < [dependencies] | [formatters]"
```

**Comment by ORESoftware** [comment]
sounds like a re-write is in the works, but the docs are woefully misinformative at the moment
